lib/pure/asyncdispatch.nim

```diff
@@ -232,6 +232,17 @@ when defined(windows) or defined(nimdoc):
     result.timers.newHeapQueue()
     result.callbacks = initDeque[proc ()](64)
 
+  proc close*(dispatcher: PDispatcher) =
+    ## Closes the Dispatcher, freeing the resources associated with it.
+    ##
+    ## Raises ``ValueError`` if the dispatcher contains registered handles,
+    ## because it indicates lack of appropriate cleanup in the application.
+    if closeHandle(dispatcher.ioPort) == 0:
+      raiseOSError(osLastError())
+    if dispatcher.handles.len != 0:
+      raise newException(ValueError,
```

Comment:
Yeah, since you're not intending for this to be caught then an AssertionError would be more appropriate. Please document this well also, emphasise that the dispatcher is going to be closed despite the AssertionError.